modyn/trainer_server/internal/trainer/remote_downsamplers/remote_rs2_downsampling.py

```diff
@@ -57,8 +57,13 @@ def _epoch_step_wr(self, target_size: int) -> None:
         self._current_subset = 0
 
     def _epoch_step_no_r(self, target_size: int) -> None:
-        max_subset = len(self._all_sample_ids) // target_size
+        if (max_subset := len(self._all_sample_ids) // target_size) == 0:
+            self._current_subset = 0
+            self._subsets = [[]]
+            return
+
```

Comment:
I am quite confused why this could happen:
This happens when `len(self._all_sample_ids)` < `target_size`. However we have `target_size = max(int(self.downsampling_ratio * len(self._all_sample_ids) / 100), 1)`. As long as `self.downsampling_ratio < 100`, which is naturally the case, the if condition here should never gets evaluated to True?

Reply:
I think this will only be the case if `len(self._all_sample_ids) == 0`.  This could happen if we have a trigger where we have fewer samples than the batch size, in which case, the data loader does not output a single batch. Then, target size is 1, and sample ids is 0.

Reply:
Ah make sense! though I am then further curious what triggering policy leads to a trigger with so few samples ;D

Reply:
it was more CGLM being very small for the first triggers :D